x-pack/plugins/lens/public/datatable_visualization/components/cell_value.tsx

```diff
@@ -5,18 +5,27 @@
  * 2.0.
  */
 
-import React, { useContext } from 'react';
+import React, { useContext, useEffect } from 'react';
 import { EuiDataGridCellValueElementProps } from '@elastic/eui';
 import type { FormatFactory } from '../../types';
 import type { DataContextType } from './types';
 
 export const createGridCell = (
   formatters: Record<string, ReturnType<FormatFactory>>,
   DataContext: React.Context<DataContextType>
-) => ({ rowIndex, columnId }: EuiDataGridCellValueElementProps) => {
-  const { table } = useContext(DataContext);
+) => ({ rowIndex, columnId, setCellProps }: EuiDataGridCellValueElementProps) => {
+  const { table, alignments } = useContext(DataContext);
   const rowValue = table?.rows[rowIndex][columnId];
   const content = formatters[columnId]?.convert(rowValue, 'html');
+  const currentAlignment = alignments && alignments[columnId];
+
+  useEffect(() => {
+    setCellProps({
+      style: {
+        textAlign: currentAlignment,
+      },
+    });
+  }, [currentAlignment, setCellProps]);
```

Comment:
The use of `useEffect` here is causing visible flickering during scrolls. I know that the docs recommend `setCellProps`, but why aren't we using class names? I did a local test using `classnames` and found that the flickering goes away.

I recorded a video of the flickering effect, and here is a single frame from the video:

<img width="559" alt="Screen Shot 2021-02-09 at 5 15 22 PM" src="https://user-images.githubusercontent.com/666475/107435747-9048f200-6afa-11eb-94b8-8d8c1574554b.png">

As you can see in this frame, the text is not evenly aligned during the scroll.

Reply:
@flash1293 It looks like you're now assigning the alignment twice: there's a className and a useEffect. Do we need both?

Reply:
@wylieconlon eh, I forgot to remove the setCellProps hook. I will fix it tomorrow morning.